### PR TITLE
Make LLVM the default backend for TE

### LINKED ISSUE #52264: [nnc][perf] 50x performance regression from `_jit_override_can_fuse_on_cpu(True)`
I am seeing some big performance regressions from two benchmarks in [TorchBench](https://github.com/pytorch/benchmark/) when I enable `torch._C._jit_override_can_fuse_on_cpu(True)`.  LearningToPaint has the performance regression both with and without freezing, while pytorch_mobilenet_v3 only has the regression when I enable freezing as well (in this case a 922x regression).

Code to reproduce:
```
import torch, re, multiprocessing
from timeit import timeit
from torchbenchmark.models import LearningToPaint, pytorch_mobilenet_v3


def measure(fuse, benchmark_module, warmup=1, number=100):
    torch.set_num_threads(1)
    torch._C._jit_override_can_fuse_on_cpu(fuse)
    name = re.sub(r"^.*[.]", "", benchmark_module.__name__)
    benchmark = benchmark_module.Model(device="cpu", jit=True)
    model, example_inputs = benchmark.get_module()
    assert isinstance(model, torch.jit.ScriptModule)

    model = model.eval()
    timeit(lambda: model(*example_inputs), number=warmup)
    print(f"    script({name:20})         = {timeit(lambda: model(*example_inputs), number=number):.3f} sec")

    model = torch.jit.freeze(model)
    timeit(lambda: model(*example_inputs), number=warmup)
    print(f"    freeze(script({name:20})) = {timeit(lambda: model(*example_inputs), number=number):.3f} sec")


for fuse in (False, True):
    print(f"_jit_override_can_fuse_on_cpu({fuse}):")
    for benchmark_module in (LearningToPaint, pytorch_mobilenet_v3):
        # Doing a subproc to ensure we aren't running cached code
        p = multiprocessing.Process(target=measure, args=(fuse, benchmark_module))
        p.start()
        p.join()
```

Output:
```
_jit_override_can_fuse_on_cpu(False):
    script(LearningToPaint     )         = 5.862 sec
    freeze(script(LearningToPaint     )) = 5.131 sec
    script(pytorch_mobilenet_v3)         = 1.089 sec
    freeze(script(pytorch_mobilenet_v3)) = 0.460 sec
_jit_override_can_fuse_on_cpu(True):
    script(LearningToPaint     )         = 275.655 sec
    freeze(script(LearningToPaint     )) = 274.557 sec
    script(pytorch_mobilenet_v3)         = 1.108 sec
    freeze(script(pytorch_mobilenet_v3)) = 424.520 sec
```

I am running on PyTorch==96fd5d87f7c739fd62b57b6cb08fd41a60d0bf40 on Ubuntu 20.10/gcc-9.3.

cc @gmagogsfm

### PR DESCRIPTION
**Pull request by navahgar**
Fixes #52264 

When CPU fusion is enabled without LLVM support in PyTorch, it causes huge slowdown (> 50x). This PR makes the LLVM backend the default backend for TE. Now, an error will be reported if CPU fusion is enabled without LLVM support, to avoid this performance regression.

This PR also updates the tests to not use LLVM, so that the old flow is continued. This is necessary because tests run in CI do not have LLVM.

### CODE REVIEW

File: torch/csrc/jit/codegen/fuser/interface.cpp [hunk 1]
```diff
@@ -43,6 +43,12 @@ bool canFuseOnGPU() {
 
 void overrideCanFuseOnCPU(bool value) {
   detail::cpu_fuser_enabled = value;
+#ifndef TORCH_ENABLE_LLVM
+  if (value) {
+    throw std::runtime_error(
```
Comment (bertmaher) [inline torch/csrc/jit/codegen/fuser/interface.cpp:48]: Does this kill the process?  I know TORCH_CHECK will propagate cleanly as a RuntimeError to python, so you might switch to that if `runtime_error` kills the python process.

**Review by bertmaher — APPROVED** [review]
One comment about the exception type but should be good to go after double checking that.

File: torch/csrc/jit/codegen/fuser/interface.cpp (hunk 1 above)
Comment (navahgar) [reply torch/csrc/jit/codegen/fuser/interface.cpp:48]: I don't see any difference even if I change to `TORCH_CHECK`. Here is the output I see:

```
Process Process-3:
Traceback (most recent call last):
  File "... python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "... python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "perf_regr.py", line 8, in measure
    torch._C._jit_override_can_fuse_on_cpu(fuse)
RuntimeError: Need LLVM support in PyTorch to enable CPU fusion!
```

Comment (navahgar) [reply torch/csrc/jit/codegen/fuser/interface.cpp]: Changed it to `TORCH_CHECK` anyways. It looks more elegant.

Comment (bertmaher) [reply torch/csrc/jit/codegen/fuser/interface.cpp]: Cool, thanks for checking!

File: test/test_jit.py [hunk 2]
```diff
@@ -7195,6 +7195,8 @@ def test(b):
             c = torch.tensor(1., requires_grad=False)
             return a, b, c  # noqa T484
 
+        torch._C._jit_override_can_fuse_on_cpu(False)
```
Comment (navahgar) [inline test/test_jit.py:7198]: @bertmaher This test was failing with my change because it was expecting flow. Hope this fix is okay.

Comment (bertmaher) [reply test/test_jit.py:7198]: Yeah, seems OK.  I'm actually not. sure why this is here in test_jit, feels like it should belong in test_jit_fuser_te.

**Review by bertmaher — APPROVED** [review]
`enable_cpu_fuser` isn't called anywhere in test_jit_fuser_te or test_tensorexpr...

**Review by bertmaher — CHANGES_REQUESTED** [review]
Meant to request changes, not accept :-p

**Comment by navahgar** [comment]
> `enable_cpu_fuser` isn't called anywhere in test_jit_fuser_te or test_tensorexpr...

Set the flag appropriately in test_jit_fuser_te and test_tensorexpr. The same change was needed in test_profiler as well.

**Review by bertmaher — APPROVED** [review]
looks good!

**Pull request closed without merging**